addon/components/paper-tabs.js

```diff
@@ -0,0 +1,308 @@
+    return htmlSafe(`${this.get('heightStyle')} ${this.get('transitionStyle')}`);
+  }),
+
+  /* Style Bindings */
+  heightStyle: computed('dynamicHeight', 'selectedTab.content.height', 'tabs.[]', 'tabsWrapper.height', function() {
+    if (this.get('dynamicHeight')) {
+      let tabsHeight = this.get('tabsWrapper.height');
+      let selectedTab = this.get('selectedTab');
+      if (selectedTab && (selectedTab.get('content.height') > 0)) {
+        return `height: ${tabsHeight + selectedTab.get('content.height')}px;`;
+      } else {
+        return `height: ${tabsHeight}px;`;
+      }
+    }
+    return '';
+  }),
+  transitionStyle: 'transition: all 0.5s cubic-bezier(0.35, 0, 0.25, 1);',
+
+  /* Logic Bindings */
+  loaded: false, // set after render
+  shouldStretchTabs: computed('stretchTabs', 'canvasWidth', function() {
+    switch (this.get('stretchTabs')) {
+      case 'always':
+        return true;
+      case 'never':
+        return false;
+      default:
+        return (!this.get('shouldPaginate') && (this.get('canvasWidth') <= 600));
+    }
+  }),
+  shouldCenterTabs: computed('centerTabs', 'shouldPaginate', function() {
+    return this.get('centerTabs') && !this.get('shouldPaginate');
+  }),
+  shouldPaginate: computed('loaded', 'noPagination', 'canvasWidth', 'pagingWidth', function() {
+    if (this.get('noPagination') || !this.get('loaded')) {
+      return false;
+    }
+    let canvasWidth = this.get('canvasWidth');
+    let pagingWidth = this.get('pagingWidth');
+    return (canvasWidth - pagingWidth) < 0;
+  }),
+  pagingWidth: computed('tabs.@each.id', 'shouldStretchTabs', function() {
+    let width = 0;
+    this.get('tabs').forEach(function(tab) {
+      if (tab.id) {
+        let element = document.getElementById(tab.id);
+        width += Math.max(element.offsetWidth, element.getBoundingClientRect().width);
+      }
+    });
+    return Math.ceil(width);
+  }),
+
+  hasContent: computed('', function() {}), // todo
+  maxTabWidth: computed('', function() {}), // todo
+  canPageBack: computed.gt('offsetLeft', 0),
+  canPageForward: computed(
+    'lastTab.offsetLeft',
+    'lastTab.offsetWidth',
+    'selectedTab.offsetLeft',
+    'selectedTab.offsetWidth',
+    'canvasWidth',
+    'offsetLeft', function() {
+    if (this.get('lastTab')) {
+      let context = this;
+      let lastTabObject = context.$(`#${context.get('lastTab.id')}`);
+      let [lastTab] = lastTabObject;
+      let pagingWidth = lastTab.offsetLeft + lastTab.clientWidth;
+      let offset = context.$('md-tabs-canvas')[0].clientWidth + context.get('offsetLeft');
+      return (pagingWidth > offset);
+    }
+  }),
+
+  /* sets the initial value as a computed property */
+  canvasWidth: 0,
+
+  didRender() {
+    let context = this;
+    run.scheduleOnce('afterRender', function() {
+      context.set('loaded', true);
+      context.set('canvasWidth', context.$().outerWidth());
+      $(window).on('resize', run.bind(context, function() {
```

Comment:
This callback is never removed from the window object, it keeps get called even after the object has been destroyed.